components/CameraView.tsx

```diff
@@ -0,0 +1,105 @@
+import { Camera, CameraCapturedPicture } from "expo-camera";
+import React, { useEffect, useRef, useState } from "react";
+import { Modal, View, TouchableOpacity, StyleSheet, Text } from "react-native";
+import { SafeAreaView } from "react-native-safe-area-context";
+import { MaterialIcons } from "@expo/vector-icons";
+import * as Haptics from "expo-haptics";
+import { CloseIconButton } from "./CloseIconButton";
+
+
+interface CameraViewProps {
+  onClose: () => void,
+  onPhotoTaken: (photo: CameraCapturedPicture) => void
+}
+
+export function CameraView(props: CameraViewProps) {
+
+  const [cameraPermissionGranted, setCameraPermissionGranted] = useState(false);
+  const [cameraReady, setCameraReady] = useState(false);
+  const [photoLoading, setPhotoLoading] = useState(false);
+  const camera = useRef<Camera>(null);
+
+  useEffect(() => {
+    Camera.requestPermissionsAsync().then(({ granted }) => {
+      if (granted) {
+        setCameraPermissionGranted(true);
+      }
+      else {
+        console.log("Nope, no camera for you");
+      }
+    })
+  }, []);
```

Comment:
Kanskje en alert med en "Ok"-knapp for å si ifra til brukeren? Akkurat nå er det ikke noe feedback

Reply:
✔